ext/fetch/wasm_streaming.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::v8;
+use deno_core::OpState;
+use deno_core::ResourceId;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+// The Wasm streaming compilation pipeline.
+pub fn handle_wasm_streaming(
+  state: Rc<RefCell<OpState>>,
+  scope: &mut v8::HandleScope,
+  value: v8::Local<v8::Value>,
+  mut wasm_streaming: v8::WasmStreaming,
+) {
+  let (url, rid) = match compile_response(scope, value) {
+    Ok(Some((url, rid))) => (url, rid),
+    Ok(None) => {
+      // 2.7
+      wasm_streaming.finish();
+      return;
+    }
+    Err(e) => {
+      // 2.8
+      let err = v8::String::new(scope, &e.to_string()).unwrap();
+      wasm_streaming.abort(Some(err.into()));
+      return;
+    }
+  };
+
+  wasm_streaming.set_url(&url);
+
+  deno_core::unsync::spawn(async move {
+    let view = deno_core::BufMutView::new(65536);
+    loop {
+      let resource = state.borrow().resource_table.get_any(rid);
+      let resource = match resource {
+        Ok(r) => r,
+        Err(_) => {
+          /* TODO(littledivy): propgate err */
+          wasm_streaming.abort(None);
+          return;
+        }
+      };
+
+      let (bytes, view) = match resource.read_byob(view).await {
+        Ok(bytes) => bytes,
+        Err(e) => {
+          println!("error reading wasm resource: {}", e);
+          /* TODO(littledivy): propgate err */
+          wasm_streaming.abort(None);
+          return;
+        }
+      };
+      if bytes == 0 {
+        break;
+      }
+
+      wasm_streaming.on_bytes_received(&view);
+    }
+
+    /* XXX: crashes here if module compilation fails */
+    wasm_streaming.finish();
```

Comment:
Addressing this in WIP feature:

https://github.com/denoland/deno_core/pull/350